### ISSUE TITLE
Issue with LightGBM

### ISSUE THREAD

**Issue by lschneidpro** [mcasl/PipeGraph#4]
Hi there,

I'm using the scikit-API of lightgbm.

1. I have been trying to fit:
2. a GBT classifier on X and y
3. concatenate X and prediction of GBT
4. re-label y with function meta()
5. a RF on X=concat and y=meta

![64472972-8da7d780-d165-11e9-838a-0789a5214518](https://user-images.githubusercontent.com/40396043/64513602-33d01a80-d2e9-11e9-911d-f71a29dc2452.jpg)

```
from pipegraph.base import PipeGraph, Concatenator
from sklearn.preprocessing import FunctionTransformer
import lightgbm as lgb
import pandas as pd



gbt = lgb.LGBMClassifier(verbose=-1)
rf = lgb.LGBMClassifier(verbose=-1, boosting_type='rf',
                        subsample=0.7,
                        colsample_bytree=0.7,
                        subsample_freq=2)
def meta(X):
    
    X[X<0]=1
    return X

meta_sk = FunctionTransformer(meta, validate=False)

concatenator = Concatenator()

steps = [('GBT', gbt),
         ('concat', concatenator),
         ('meta', meta_sk),
         ('RF', rf),
         ]

connections = { 'GBT': {'X': 'X',
                          'y': 'y'},
                'concat': {'X1': 'GBT',
                           'X2': 'X'},
                'meta': {'X': 'y'},
                'RF': {'X': 'concat',
                         'y': 'meta'},
                }

pgraph = PipeGraph(steps=steps, fit_connections=connections)

pgraph.fit(X=df_cv[xvars].to_numpy(),y=df_cv[y_gbt].to_numpy().reshape(-1))
```

However I got the following error. I assume y is wrongly passed to GBT. Please let me know :)
```
[LightGBM] [Warning] Unknown parameter: y
ERROR: step.fit call ValueError!
ERROR: _fit.predict call ValueError!
Out[85]: 
PipeGraph(fit_connections={'GBT': {'X': 'X', 'y': 'y'},
                           'RF': {'X': 'concat', 'y': 'meta'},
                           'concat': {'X1': 'GBT', 'X2': 'X'},
                           'meta': {'X': 'y'}},
          log_level=None,
          predict_connections={'GBT': {'X': 'X', 'y': 'y'},
                               'RF': {'X': 'concat', 'y': 'meta'},
                               'concat': {'X1': 'GBT', 'X2': 'X'},
                               'meta': {'X': 'y'}},
          steps=[('GBT',
                  LGBMClassifier(boosting_type='gbdt', class_weight=None,
                                 colsample_b...
                  LGBMClassifier(boosting_type='rf', class_weight=None,
                                 colsample_bytree=0.7, importance_type='split',
                                 learning_rate=0.1, max_depth=-1,
                                 min_child_samples=20, min_child_weight=0.001,
                                 min_split_gain=0.0, n_estimators=100,
                                 n_jobs=-1, num_leaves=31, objective=None,
                                 random_state=None, reg_alpha=0.0,
                                 reg_lambda=0.0, silent=True, subsample=0.7,
                                 subsample_for_bin=200000, subsample_freq=2,
                                 verbose=-1))])
```

**Comment by mcasl** [comment]
Hi,
I've built a unit test around this issue to debug this behaviour, see the code below.
I've tried to reproduce the error but everything seems fine on the pipegraph side.

Given these artificial datasets the output equals to np.zeros() . I'm going to investigate the behaviour of lgdb regarding the names of the columns, since currently Concatenator may duplicate column names. That may be a reason for the np.zeros().

Anyway, I'd welcome if you could specify if these artificial datasets are representative of the one you are working with and causing you problems. Otherwise, it would be nice if you could write a couple of lines recreating a representative toy dataset .
 
As a side note, it seems to me that you are interested in adding 'predict_proba' according to your drawing, as in    `concat': {'X1': ('GBT', 'predict_proba'),`

```
import unittest
import numpy as np
import pandas as pd
from pipegraph.base import PipeGraph, Concatenator
from sklearn.preprocessing import FunctionTransformer
from sklearn.datasets import make_blobs
import lightgbm as lgb


class TestUserIssues(unittest.TestCase):
    def setUp(self):
        self.size = 40
        data = make_blobs(n_samples=self.size, n_features=4)
        self.X = data[0]
        self.y = data[1]

    def test__issue_with_lgbm(self):
        gbt = lgb.LGBMClassifier(verbose=-1)
        rf = lgb.LGBMClassifier(verbose=-1, boosting_type='rf',
                                subsample=0.7,
                                colsample_bytree=0.7,
                                subsample_freq=2)

        def meta(X):
            X[X < 0] = 1
            return X

        meta_sk = FunctionTransformer(meta, validate=False)

        concatenator = Concatenator()

        steps = [('GBT', gbt),
                 ('concat', concatenator),
                 ('meta', meta_sk),
                 ('RF', rf),
                 ]

        connections = {'GBT': {'X': 'X',
                               'y': 'y'},
                       'concat': {'X1': ('GBT', 'predict_proba'),
                                  'X2': 'X'},
                       'meta': {'X': 'y'},
                       'RF': {'X': 'concat',
                              'y': 'meta'},
                       }

        pgraph = PipeGraph(steps=steps, fit_connections=connections)

        pgraph.fit(X=self.X, y=self.y)
      

if __name__ == '__main__':
    unittest.main()
```
